### PR TITLE
Migrate `LaunchDarklyConfigstore`

### LINKED ISSUE #519: Migrate `LaunchDarklyConfigStore` to `@configu/integrations`
### Suggestion

Migrate the `LaunchDarklyConfigStore` from its current setup to the new `@configu/integrations` module as part of the major restructuring. The goal is to move all third-party integrations, including stores, expressions, and libraries, to `@configu/integrations`.

### Motivation

This migration is part of a larger effort to better organize and modularize the product. Moving the `LaunchDarklyConfigStore` will align with the new structure, making the system cleaner and more maintainable.

### Context

You can refer to the already migrated stores for guidance:  
[Migrated stores in integrations](https://github.com/configu/configu/tree/main/packages/integrations/stores).

Base stores implementations from `@configu/ts` can be found here:  
[Base stores in TS](https://github.com/configu/configu/tree/52cee9c41fb03addc4c0983028e37df42945f5b7/packages/ts/src/stores).

The old node-based stores were previously in `@configu/node`:  
[Node stores](https://github.com/configu/configu/tree/52cee9c41fb03addc4c0983028e37df42945f5b7/packages/node/src/stores).

The old browser-based stores were previously in `@configu/browser`:  
[Browser stores](https://github.com/configu/configu/tree/52cee9c41fb03addc4c0983028e37df42945f5b7/packages/browser/src/stores).

### PR DESCRIPTION
**Pull request by TomerPacific**
Closes #519 

This pull request introduces a new integration for LaunchDarkly within the Configu ecosystem. The changes cover the addition of a new package, its configuration, and necessary dependencies.

### New Integration for LaunchDarkly:

* [`packages/integrations/stores/launch-darkly/README.md`](diffhunk://#diff-834e8e2154db4cb16e6a28a7a4b2d70fecb6b2c16f04b3053158b254efa9150cR1-R42): Added documentation for the new LaunchDarkly integration, including configuration and usage examples.
* [`packages/integrations/stores/launch-darkly/package.json`](diffhunk://#diff-037a0f0e4f860bc2f67eedbf3a1949f1eac236d97fcc1cb3e96f6a64e7001b90R1-R36): Created a new package configuration for the LaunchDarkly integration, specifying dependencies, scripts, and metadata.
* [`packages/integrations/stores/launch-darkly/src/launch-darkly.ts`](diffhunk://#diff-527c61193dabd0e51f8b9c3bbea78a38adf116ca44dd740728ad468878800defR1-R34): Implemented the LaunchDarkly configuration store with necessary imports and a class to handle configuration queries.

### Configuration Files:

* [`packages/integrations/stores/launch-darkly/tsconfig.json`](diffhunk://#diff-b93a45f22689a5bcd13965935090a48a38e608c6fb4670afd141c2cb1129af50R1-R3): Extended the base TypeScript configuration for the new package.
* [`packages/integrations/stores/launch-darkly/tsup.config.ts`](diffhunk://#diff-3f31066948fc5f1e12a96c765ca2930114dba46749cf66e68a85636e2e6949d8R1-R21): Defined the build configuration using `tsup` for the new package.

### Dependency Management:

* [`pnpm-lock.yaml`](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR678-R705): Updated to include the new dependencies and devDependencies for the LaunchDarkly integration package.

### CODE REVIEW

File: packages/integrations/stores/launch-darkly/src/launch-darkly.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { type EvaluationContext, OpenFeature } from '@openfeature/server-sdk';
+import { LaunchDarklyProvider } from '@launchdarkly/openfeature-node-server';
+import { type LDOptions } from '@launchdarkly/node-server-sdk';
+import { type ConfigQuery, type Config } from '@configu/sdk';
+import _ from 'lodash';
```
Comment (RichardAkman) [inline packages/integrations/stores/launch-darkly/src/launch-darkly.ts:5]: Please change the lodash dependency from `lodash` to `lodash-es` and also make sure it's present in the `package.json` dependencies

File: packages/integrations/stores/launch-darkly/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# @configu-integrations/launchdarkly
```
Comment (RichardAkman) [inline packages/integrations/stores/launch-darkly/README.md:1]:
```suggestion
# @configu-integrations/launch-darkly
```

File: packages/integrations/stores/launch-darkly/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+Integrates the Configu Orchestrator with [LaunchDarkly](https://launchdarkly.com/).
+
+- Name: LaunchDarkly
+- Category:
```
Comment (RichardAkman) [inline packages/integrations/stores/launch-darkly/README.md:6]:
```suggestion
- Category: Feature flag manager
```

File: packages/integrations/stores/launch-darkly/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+## Configuration
+
+Configu needs to be authorized to access
```
Comment (RichardAkman) [inline packages/integrations/stores/launch-darkly/README.md:10]:
```suggestion
Configu needs to be authorized to access your LaunchDarkly project. You must specify an `sdkKey` that corresponds to an SDK key of an environment of some LaunchDarkly project and a context that has a `key` or a `targetingKey` that identifies the subject (end-user, or client service) of a flag evaluation. You may also configure [`ldOptions`](https://launchdarkly.github.io/node-server-sdk/interfaces/_launchdarkly_node_server_sdk_.LDOptions.html) to customize the LaunchDarkly client.
```

File: packages/integrations/stores/launch-darkly/README.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+## Usage
+
+### `.configu` store declaration
+
+```yaml
+stores:
+  my-store:
+    type: launch-darkly
+    configuration:
```
Comment (RichardAkman) [inline packages/integrations/stores/launch-darkly/README.md:20]:
```suggestion
    configuration:
      sdkKey: example-sdkKey
        context:
          targetingKey: default
```

File: packages/integrations/stores/launch-darkly/README.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+### CLI examples
+
+#### Upsert command
+
+```bash
+configu upsert --store "my-store" --set "test" --schema "./start.cfgu.json" \
+    -c "GREETING=hey" \
+    -c "SUBJECT=configu node.js sdk"
```
Comment (RichardAkman) [inline packages/integrations/stores/launch-darkly/README.md:30]:
```suggestion
    -c "SUBJECT=configu"
```

File: packages/integrations/stores/launch-darkly/README.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+#### Eval and export commands
+
+```bash
+configu eval --store "my-store" --set "test" --schema "./start.cfgu.json" \
+ | configu export
+```
+
+## References
+
+- Integration documentation: https://docs.launchdarkly.com/home/getting-started/setting-up?_gl=1*11fhv7o*_gcl_au*MTQ4NDYyNzQ0OC4xNzI5ODgzMTI5
```
Comment (RichardAkman) [inline packages/integrations/stores/launch-darkly/README.md:42]:
```suggestion
- Integration documentation: https://docs.launchdarkly.com
```
The link to be to the root page of the docs of the integration.

**Review by RichardAkman — CHANGES_REQUESTED** [review]
Great work, review my minor comments and apply my suggestions to flesh out this PR and make it ready for approval 🙏🏻 ⭐ 🚀 

Please also make sure to resolve the conflict of the pnpm-lock file.

File: packages/integrations/stores/launch-darkly/README.md (hunk 7 above)
Comment (TomerPacific) [reply packages/integrations/stores/launch-darkly/README.md]: Done.

File: packages/integrations/stores/launch-darkly/README.md (hunk 6 above)
Comment (TomerPacific) [reply packages/integrations/stores/launch-darkly/README.md]: Done.

File: packages/integrations/stores/launch-darkly/README.md (hunk 5 above)
Comment (TomerPacific) [reply packages/integrations/stores/launch-darkly/README.md:20]: Done.

File: packages/integrations/stores/launch-darkly/README.md (hunk 4 above)
Comment (TomerPacific) [reply packages/integrations/stores/launch-darkly/README.md]: Done.

File: packages/integrations/stores/launch-darkly/README.md (hunk 3 above)
Comment (TomerPacific) [reply packages/integrations/stores/launch-darkly/README.md]: Done.

File: packages/integrations/stores/launch-darkly/README.md (hunk 2 above)
Comment (TomerPacific) [reply packages/integrations/stores/launch-darkly/README.md]: Done.

File: packages/integrations/stores/launch-darkly/src/launch-darkly.ts (hunk 1 above)
Comment (TomerPacific) [reply packages/integrations/stores/launch-darkly/src/launch-darkly.ts]: Done.

**Comment by TomerPacific** [comment]
@RichardAkman - I have gone over all the changes you requested and fixed the conflict in the lock file. Let me know if anything else is necessary.

File: packages/integrations/stores/launch-darkly/README.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+### `.configu` store declaration
+
+```yaml
+stores:
+  my-store:
+    type: launch-darkly
+    configuration:
+      sdkKey: example-sdkKey
+        context:
```
Comment (davidmeirlevy) [inline packages/integrations/stores/launch-darkly/README.md:22]: is that the correct indentation?
it seems like "context" and "sdkKey" should be siblings.

**Review by davidmeirlevy — CHANGES_REQUESTED** [review]
@TomerPacific thanks for your PR.

I wrote one small comment, but everything else looks fine.

File: packages/integrations/stores/launch-darkly/README.md (hunk 8 above)
Comment (RichardAkman) [reply packages/integrations/stores/launch-darkly/README.md:22]: It's correct, that's how the underlying SDK expects to get the parameter

**Review by RichardAkman — APPROVED** [review]
LGTM, thank you for your contribution ⭐

**Review by davidmeirlevy — APPROVED** [review]

**Comment by pelegpor** [comment]
Hi @TomerPacific ,

Here are the details for claiming your Hacktoberfest prizes! 🎉 Check out [#465  (comment)](https://github.com/configu/configu/issues/465#issuecomment-2498550259) for more information or simply fill out the form here: [Claim Your Swag](https://share-eu1.hsforms.com/2SVc5VKY3Thi04Bvjh17WKw2e62o0).  

Remember, you only need to fill out the form once for all your PRs. 😊

**Pull request merged**